File: src/guepardoapps/guepardonotes/customadapter/NoteListAdapter.java
package guepardoapps.guepardonotes.customadapter;

import java.util.ArrayList;

import android.annotation.SuppressLint;
import android.content.Context;
import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.view.View.OnClickListener;
import android.widget.BaseAdapter;
import android.widget.TextView;

import guepardoapps.guepardonotes.R;
import guepardoapps.guepardonotes.activities.ActivityDetails;
import guepardoapps.guepardonotes.common.Constants;
import guepardoapps.guepardonotes.model.Note;

import guepardoapps.toolset.services.NavigationService;

public class NoteListAdapter extends BaseAdapter {

	private Context _context;
	private NavigationService _navigationService;

	private ArrayList<Note> _notes;

	private static LayoutInflater _inflater = null;

	public NoteListAdapter(Context context, ArrayList<Note> notes) {
		_context = context;
		_navigationService = new NavigationService(_context);

		_notes = notes;

		_inflater = (LayoutInflater) _context.getSystemService(Context.LAYOUT_INFLATER_SERVICE);
	}

	@Override
	public int getCount() {
		return _notes.size();
	}

	@Override
	public Object getItem(int position) {
		return position;
	}

	@Override
	public long getItemId(int position) {
		return position;
	}

	public class Holder {
		TextView _title;
	}

	@SuppressLint({ "InflateParams", "ViewHolder" })
	@Override
	public View getView(final int index, View convertView, ViewGroup parent) {
		Holder holder = new Holder();
		View rowView = _inflater.inflate(R.layout.list_item, null);

		holder._title = (TextView) rowView.findViewById(R.id.itemTitle);
		holder._title.setText(_notes.get(index).GetTitle());
		holder._title.setOnClickListener(new OnClickListener() {
			@Override
			public void onClick(View v) {

				Bundle details = new Bundle();
				details.putInt(Constants.BUNDLE_NOTE_ID, _notes.get(index).GetId());
				details.putString(Constants.BUNDLE_NOTE_TITLE, _notes.get(index).GetTitle());
				details.putString(Constants.BUNDLE_NOTE_CONTENT, _notes.get(index).GetContent());

				_navigationService.NavigateWithData(ActivityDetails.class, details, true);
			}
		});

		return rowView;
	}
}

File: mobile/src/main/java/guepardoapps/mynote/customadapter/NoteListAdapter.kt
package guepardoapps.mynote.customadapter

import android.annotation.SuppressLint
import android.content.Context
import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.*
import com.afollestad.materialdialogs.MaterialDialog
import com.rey.material.widget.FloatingActionButton
import guepardoapps.mynote.R
import guepardoapps.mynote.activities.ActivityEdit
import guepardoapps.mynote.controller.NavigationController
import guepardoapps.mynote.database.note.DbNote
import guepardoapps.mynote.model.Note

@ExperimentalUnsignedTypes
internal class NoteListAdapter(private val context: Context, private val list: Array<Note>) : BaseAdapter() {

    private val navigationController: NavigationController = NavigationController(context)

    private val inflater: LayoutInflater = context.getSystemService(Context.LAYOUT_INFLATER_SERVICE) as LayoutInflater

    private class Holder {
        lateinit var title: TextView
        lateinit var content: TextView
        lateinit var dateTime: TextView
        lateinit var edit: FloatingActionButton
        lateinit var delete: FloatingActionButton
    }

    override fun getItem(position: Int): Note = list[position]

    override fun getItemId(position: Int): Long = position.toLong()

    override fun getCount(): Int = list.size

    @SuppressLint("SetTextI18n", "ViewHolder", "InflateParams")
    override fun getView(index: Int, convertView: View?, parentView: ViewGroup?): View {
        val rowView: View = inflater.inflate(R.layout.list_item, null)

        Holder().apply {
            val note = list[index]

            title = rowView.findViewById(R.id.itemTitle)
            content = rowView.findViewById(R.id.itemContent)
            dateTime = rowView.findViewById(R.id.itemDateTime)

            edit = rowView.findViewById(R.id.btnEdit)
            delete = rowView.findViewById(R.id.btnDelete)

            title.text = note.title
            content.text = note.content
            dateTime.text = "${note.dateString} / ${note.timeString}"

            edit.setOnClickListener {
                val bundle = Bundle()
                bundle.putString(context.getString(R.string.bundleDataId), note.id)
                navigationController.navigateWithData(ActivityEdit::class.java, bundle, false)
            }

            delete.setOnClickListener {
                MaterialDialog(context).show {
                    title(text = context.getString(R.string.delete))
                    message(text = String.format(context.getString(R.string.deleteRequest), note.title))
                    positiveButton(text = context.getString(R.string.yes)) { DbNote(context).delete(note.id) }
                    negativeButton(text = context.getString(R.string.no))
                }
            }
        }

        return rowView
    }
}